GVFS/GVFS.Common/MacProductUpgraderInfo.cs

```diff
@@ -0,0 +1,27 @@
+using GVFS.Common.FileSystem;
+using GVFS.Common.Tracing;
+using System;
+using System.IO;
+
+namespace GVFS.Common
+{
+    public class MacProductUpgraderInfo : ProductUpgraderInfoImpl

```

Comment:
Again in this PR, I don't think we should have platform specific code in `GVFS.Common`.

Reply:
Yes - they currently share the same pattern. At a higher level, I think the question is do we want upgrader specific code in the platform, or the upgrader project to have knowledge of the platforms.

How would you have the upgrade code load the correct instance type (based on platform)?

Reply:
I am going to look at moving this to the platform specific projects